sources/packages/backend/apps/queue-consumers/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from "@nestjs/common";
+import { StudentAssessment } from "@sims/sims-db";
+import { Repository } from "typeorm";
+import { InjectRepository } from "@nestjs/typeorm";
+
+/**
+ * Manages the student assessment related operations
+ * for the queue consumers application.
+ */
+@Injectable()
+export class StudentAssessmentService {
+  constructor(
+    @InjectRepository(StudentAssessment)
+    private readonly studentAssessmentRepo: Repository<StudentAssessment>,
+  ) {}
+
+  /**
+   * Gets the student assessment by its id.
+   * @param assessmentId assessment id.
+   * @returns assessment found, otherwise, null.
+   */
+  async getAssessmentById(assessmentId: number): Promise<StudentAssessment> {
```

Comment:
As it is being used to rollback the over awards, should be not add an optional filter to check for application status to be cancelled?

Reply:
I will add some validations.